modules/widgets/googleplus-badge.php

```diff
@@ -0,0 +1,309 @@
+	private $allowed_themes      = array('light', 'dark');
+	private $default_layout      = 'portrait';
+	private $allowed_layouts     = array('landscape', 'portrait');
+	private $default_type        = 'person';
+	private $allowed_types       = array();
+
+	function __construct() {
+		$this->min_width = min( $this->min_width_portrait, $this->min_width_landscape );
+		$this->allowed_types = array(
+			'person'    => __('Person Widget', 'jetpack'),
+			'page'      => __('Page Widget', 'jetpack'),
+			'community' => __('Community Widget', 'jetpack'),
+		);
+
+		/**
+		 * Modify widget name.
+		 *
+		 * This filter is mainly used to add the "(Jetpack)" suffix to the widget names.
+		 *
+		 * @param string $widget_name The name of widget
+		 */
+		parent::__construct(
+			'googleplus-badge',
+			apply_filters( 'jetpack_widget_name', __( 'Google+ Badge', 'jetpack' ) ),
+			array(
+				'classname'   => 'widget_googleplus_badge',
+				'description' => __( 'Display a Google+ Badge to connect visitors to your Google+', 'jetpack' )
+			)
+		);
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_scripts' ) );
+
+		if ( is_active_widget( '', '', 'googleplus-badge' ) ) {
+			add_action( 'wp_print_styles',   array( $this, 'enqueue_script' ) );
+			add_filter( 'script_loader_tag', array( $this, 'replace_script_tag' ), 10, 2 );
+		}
+	}
+
+	function enqueue_script() {
+		wp_enqueue_script( 'googleplus-widget', 'https://apis.google.com/js/platform.js' );
+	}
+
+	function replace_script_tag( $tag, $handle ) {
+		if ( 'googleplus-widget' !== $handle ) {
+			return $tag;
+		}
+
+		return str_replace( ' src', ' async defer src', $tag );
+	}
+
+	function enqueue_admin_scripts() {
+		global $pagenow;
+
+		if ( 'widgets.php' == $pagenow || 'customize.php' == $pagenow ) {
+			wp_enqueue_script( 'googleplus-widget-admin', plugins_url( '/google-plus/js/admin.js', __FILE__ ), array( 'jquery' ) );
+		}
+	}
+
+	function widget( $args, $instance ) {
+		if ( empty( $instance['href'] ) || ! $this->is_valid_googleplus_url( $instance['href'] ) ) {
+			if ( current_user_can('edit_theme_options') ) {
+				echo $args['before_widget'];
+				echo '<p>' . sprintf(
+					__( 'It looks like your Google+ URL is incorrectly configured. Please check it in your <a href="%s">widget settings</a>.', 'jetpack' ),
+					admin_url( 'widgets.php' )
+				) . '</p>';
+				echo $args['after_widget'];
+			}
+			echo '<!-- Invalid Google+ URL -->';
+			return;
+		}
+
+
+		/** This filter is documented in core/src/wp-includes/default-widgets.php */
+		$title = apply_filters( 'widget_title', $instance['title'] );
+
+		if ( empty( $title ) ) {
+			$title = esc_html__( 'Google+', 'jetpack' );
+		}
+
+		echo $args['before_widget'];
+		echo $args['before_title'] . esc_html( $title ) . $args['after_title'];
+
+		switch( $instance['type'] ) {
+			case 'person':
+			case 'page':
+				printf(
+					'<g:%s href="%s" layout="%s" theme="%s" showcoverphoto="%s" showtagline="%s" width="%s"></g:%s>',
```

Comment:
Let's go with the HTML5 versions.  Our theme team requests to have `overflow: hidden;` applied to the container.  This is easy with HTML5